src/sentry/workflow_engine/handlers/condition/group_event_handlers.py

```diff
@@ -116,3 +117,50 @@ def evaluate_value(job: WorkflowJob, comparison: Any) -> bool:
         elif desired_match == MatchType.LESS_OR_EQUAL:
             return level <= desired_level
         return False
+
+
+@condition_handler_registry.register(Condition.TAGGED_EVENT)
+class TaggedEventConditionHandler(DataConditionHandler[WorkflowJob]):
+    @staticmethod
+    def evaluate_value(job: WorkflowJob, comparison: Any) -> bool:
+        event = job["event"]
+        raw_tags = event.tags
+        key = comparison.get("key")
+        match = comparison.get("match")
+        value = comparison.get("value")
+
+        if not key or not match:
+            return False
+
+        key = key.lower()
+
+        tag_keys = (
+            k
+            for gen in (
+                (k.lower() for k, v in raw_tags),
+                (tagstore.backend.get_standardized_key(k) for k, v in raw_tags),
+            )
+            for k in gen
+        )
+
+        # NOTE: IS_SET condition differs btw tagged_event and event_attribute so not handled by match_values
+        if match == MatchType.IS_SET:
+            return key in tag_keys
+
+        elif match == MatchType.NOT_SET:
```

Comment:
i'm not following why we can't use the `match_values` function for these if we have a comma separated string? similarly for `EventAttribute`, i just noticed